### PR TITLE
Fix c2e's Kafka setup when using an external cluster

### PR DESCRIPTION
**Pull request by refi64**
The credentials were always assumed to be those for the example one.

Signed-off-by: Ryan Gonzalez <ryan.gonzalez@collabora.com>

### CODE REVIEW

**Comment by calohmn** [comment]
@refi64 Thanks for the PR. In general, chart changes have to be accompanied with an increase of the chart version in the corresponding `Chart.yaml`. In the case of the cloud2edge chart that also applies to the `appVersion` (setting both to `0.2.5` then).

File: packages/cloud2edge/post-install/hono-kafka-connection.json [hunk 1]
```diff
@@ -151,8 +157,14 @@
         }
       ],
       "specificConfig": {
+        {{- if .Values.hono.kafkaMessagingClusterExample.enabled }}
         "saslMechanism": "plain",
         "bootstrapServers": "{{ .Release.Name }}-kafka:9092",
+        {{- else }}
+        {{ $kafka := .Values.hono.adapters.kafkaMessagingSpec.commonClientConfig }}
+        "saslMechanism": "{{ (index $kafka "sasl.mechanism") | lower }}",
```
Comment (calohmn) [inline packages/cloud2edge/post-install/hono-kafka-connection.json:165]: Could you use "plain" as fallback here? The "sasl.mechanism" property might not have been set in `commonClientConfig`.

**Comment by refi64** [comment]
@calohmn oops, version updated!

**Review by calohmn — APPROVED** [review]
LGTM

Thanks for contributing :+1:

**Pull request merged**